client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -204,6 +209,10 @@ private EventRead<Type> createEmptyEvent(String checkpoint) {
         return new EventReadImpl<>(null, refreshAndGetPosition(), null, checkpoint);
     }
 
+    private EventRead<Type> createEmptyEventEndOfStream(Boolean isEndOfStream) {
+        return new EventReadImpl<>(null, refreshAndGetPosition(), null, null, isEndOfStream);
+    }
```

Comment:
Given that we are calling this only when end of stream has been reached and `isEndOfStream` is true, do we need this value to be passed as a parameter?

Reply:
Right, this can be done